blocks/bricks/sequence_generators.py

```diff
@@ -94,36 +94,43 @@ class BaseSequenceGenerator(Initializable):
 
     """
     @lazy
-    def __init__(self, readout, transition, fork=None, **kwargs):
+    def __init__(self, readout, transition, fork, **kwargs):
         super(BaseSequenceGenerator, self).__init__(**kwargs)
         self.readout = readout
         self.transition = transition
         self.fork = fork
 
-        self.state_names = transition.compute_states.outputs
-        self.context_names = transition.apply.contexts
-        self.glimpse_names = transition.take_glimpses.outputs
         self.children = [self.readout, self.fork, self.transition]
 
+    @property
```

Comment:
I'm not sure what you mean by private? I just made them properties because right now the user was able to change them, which would just lead to errors because they wouldn't match up with the transition inputs anymore. It seemed like they were just used as shorthands more than anything.